### PR TITLE
Currency hotfixes

### PR DESCRIPTION
**Pull request by zorun**
Hardcode list of currencies, and temporarily disable some currency operations to prevent crashes.

Here is what is disabled:

- setting or changing the default currency on an existing project
- adding or editing a bill with a currency that differs from the default currency of the project

Adjust and skip tests as necessary.

The real fix will come after #1232  is discussed

### CODE REVIEW

File: ihatemoney/tests/api_test.py [hunk 1]
```diff
@@ -615,6 +617,7 @@ def test_bills_with_calculation(self):
             )
             self.assertStatus(400, req)
 
+    @pytest.mark.skip(reason="Currency conversion is broken")
```
Comment (almet) [inline ihatemoney/tests/api_test.py:620]: Yay, we benefit from new pytest markers :-)

**Comment by zorun** [comment]
I'd like to release version 6.1.1 as soon as this is merged (which means really soon)

File: ihatemoney/tests/budget_test.py [hunk 2]
```diff
@@ -1703,7 +1706,7 @@ def test_rss_feed(self):
         Tests that the RSS feed output content is expected.
         """
         with fake_time("2023-07-25 12:00:00"):
-            self.post_project("raclette")
+            self.post_project("raclette", default_currency="EUR")
```
Comment (almet) [inline ihatemoney/tests/budget_test.py:1709]: Not sure why we need to specify the currency here?

**Review by almet — COMMENTED** [review]
Hi, thanks for doing this.

What about adding a test scenario which reproduces the problems we have currently? So we can ensure this is fixed by the changes you're doing here?

If it's too long to do, I think we should merge this and release a bugfix release, but I think it could be a nice addition.

File: ihatemoney/tests/main_test.py [hunk 3]
```diff
@@ -385,9 +385,9 @@ def test_only_one_instance(self):
         assert one == two
 
     def test_get_currencies(self):
-        assert set(self.converter.get_currencies()) == set(
-            ["USD", "EUR", "CAD", "PLN", CurrencyConverter.no_currency]
-        )
+        currencies = self.converter.get_currencies()
+        for currency in ["USD", "EUR", "CAD", "PLN", CurrencyConverter.no_currency]:
```
Comment (almet) [inline ihatemoney/tests/main_test.py:389]: I'm a bit confused with `CurrencyConverter.no_currency` here. Should it be `XXX`?

Comment (zorun) [reply ihatemoney/tests/main_test.py:389]: Well yes, it's the same thing.

File: ihatemoney/tests/budget_test.py (hunk 2 above)
Comment (almet) [reply ihatemoney/tests/budget_test.py:1709]: Okay, thanks for the heads up.

**Comment by zorun** [comment]
You mean simulating the exception that would be raised by the API call? It's a good idea, that would have allowed us to catch this way earlier. But I'm not motivated enough to do this now...

File: ihatemoney/tests/main_test.py (hunk 3 above)
Comment (almet) [reply ihatemoney/tests/main_test.py:389]: [Heh, yes it is](https://github.com/spiral-project/ihatemoney/pull/1240/files#diff-9534886c8df967ba5f34f0c976eb6790ec62b2e39103822885bff66ad95a2d4cR19)

**Comment by almet** [comment]
> You mean simulating the exception that would be raised by the API call? It's a good idea, that would have allowed us to catch this way earlier. But I'm not motivated enough to do this now...

I understand, especially as it's a quick fix for an emergency release. Let's move on then, I won't have the time to add it either.

Thanks again for doing that!

**Comment by zorun** [comment]
Thanks for the review! I will make the release now.

**Pull request merged**